waku/waku_sync/storage.nim

```diff
@@ -0,0 +1,59 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import std/[times, tables], chronicles, chronos, stew/results
+
+import ./raw_bindings
+
+logScope:
+  topics = "waku sync"
+
+type WakuSyncStorageManager* = object
+  storages: OrderedTable[string, Storage]  # Map of dateTime and Storage objects. DateTime is of the format YYYYMMDDHH
+  maxHours: times.Duration
+
+proc new*(T: type WakuSyncStorageManager, hoursToStore: times.Duration = initDuration(minutes=120)): T =
+  return WakuSyncStorageManager(maxHours:hoursToStore.inHours)
+
+#Time should be in YYYYMMDDHH format.
+proc createStorage(
+    self: WakuSyncStorageManager, time: string
+): Result[Storage, string] =
+  let storage:Storage = Storage.new().valueOr:
+    error "storage creation failed"
+    return err("storage creation failed")
+  self.storages[time] = storage
+  return storage
+
+proc getRecentStorage(): Result[Storage, string]=
+
+
+#Time should be in YYYYMMDDHH format.
+proc getStorage(self: WakuSyncStorageManager, time: string): Result[Storage, string] =
+    if self.storages.hasKey(time):
+        return self.storages[time]
+  return err("no storage found for given dateTime")
+
+proc retrieveStorage*(
+    self: WakuSyncStorageManager, time: int64
+): Result[Storage, string] =
+  let unixf = times.fromUnixFloat(float(time))
+
+  let dateTime = times.format(unixf, "YYYYMMDDHH")
```

Comment:
Lets do the opposite. Use a timestamp internally but allow fetching by date & time formatted string.

Reply:
I was able to retrieve storage by using timestamp. I don't think we need to expose any other way and keep the string part hidden within. I did not want to use timestamp as the granularity with which we want to maintain storage is in terms of hours and timestamp is in seconds.